scylla-cql/src/frame/request/auth_response.rs

```diff
@@ -1,37 +1,22 @@
 use crate::frame::frame_errors::ParseError;
 use bytes::BufMut;
-use std::convert::TryInto;
 
 use crate::frame::request::{Request, RequestOpcode};
-use crate::frame::Authenticator;
 
 // Implements Authenticate Response
 pub struct AuthResponse {
-    pub username: Option<String>,
-    pub password: Option<String>,
-    pub authenticator: Authenticator,
+    pub response: Option<Vec<u8>>,
+    pub authenticator: String,
 }
 
 impl Request for AuthResponse {
     const OPCODE: RequestOpcode = RequestOpcode::AuthResponse;
 
     fn serialize(&self, buf: &mut impl BufMut) -> Result<(), ParseError> {
-        if self.username.is_none() || self.password.is_none() {
-            return Err(ParseError::BadDataToSerialize(
-                "Bad credentials: username or password missing. You can use SessionBuilder::user(\"user\", \"pass\") to provide credentials.".to_string(),
-            ));
+        if let Some(bytes) = self.response.as_ref() {
+            buf.put_slice(bytes.as_slice());
```

Comment:
The `AUTH_RESPONSE` field has a single field which is serialized as `[bytes]`. The `[bytes]` is a concept from the CQL protocol description an is semantically equivalent to `Option<Vec<u8>>`.

What you did above is not how you serialize `[bytes]`. Please refer to the CQL protocol spec on how it should be serialized.

https://github.com/scylladb/scylla-tools-java/blob/5f2b91d774843c343e9fb07c84fb6298bd07a4e9/doc/native_protocol_v4.spec